src/transformers/modeling_utils.py

```diff
@@ -1006,6 +1006,72 @@ def get_input_embeddings(self) -> nn.Module:
         else:
             raise NotImplementedError
 
+    def warn_if_pad_token_in_input_ids_no_attention_mask(self, input_ids, attention_mask):
+        if attention_mask is not None:
+            # if the attention_mask is defined it's all good
+            return
+
+        if not hasattr(self, "warnings_issued"):
+            self.warnings_issued = {}
+
+        if self.warnings_issued.get("pad_token_in_input_ids", False):
+            # if warning has already been thrown don't throw again
+            return
+
+        is_pad_token_in_input_ids = self.config.pad_token_id is not None and self.config.pad_token_id in input_ids
+        if is_pad_token_in_input_ids:
+            # things become tricky if <pad> is equal to either BOS or EOS:
+            # in this case we cannot reasonably know whether the user should use an attention_mask or not
+            # because a in left padding both <pad> and BOS could be in the beginning
+            # and in right padding both <pad> and EOS could be in the end.
+            # Both cases look the same for `input_ids`, but one is correct behavior, the other one incorrect.
+            # In this case, we should still throw a warning because most models don't have
+            # <pad> == EOS or <pad> == BOS or <pad> == SEP.
+            is_pad_token_equal_to_bos_token = (
+                self.config.bos_token_id is not None and self.config.bos_token_id == self.config.pad_token_id
+            )
+            is_pad_token_equal_to_eos_token = (
+                self.config.eos_token_id is not None and self.config.eos_token_id == self.config.pad_token_id
+            )
+            is_pad_token_equal_to_sep_token = (
+                self.config.sep_token_id is not None and self.config.sep_token_id == self.config.pad_token_id
+            )
+
+            warn_string = (
+                f"The input IDs {input_ids} contains the `pad_token_id` {self.config.pad_token_id}, "
+                "but NO `attention_mask` is passed."
+            )
+
+            if is_pad_token_equal_to_bos_token:
+                warn_string += (
+                    "\nWe strongly recommend passing an `attention_mask` to avoid possibly incorrectly computing the"
+                    " attention weights. \nYou can ignore this warning, if your `pad_token_id`"
+                    f" {self.config.pad_token_id} is identical to your `bos_token_id` {self.config.bos_token_id} AND"
+                    " your input is NOT padded."
+                )
```

Comment:
Maybe here let's check if the pad token ID is only used once per input at the beginning before throwing the warning?